### PR TITLE
feat: `prql` namespace for polars.DataFrame and polars.LazyFrame

### LINKED ISSUE #371: `prql` namespace for `polars.DataFrame` and `polars.LazyFrame`
Works like the current `pyprql.pandas_accessor`.

I would like to wrap this pola-rs/polars#15783 (included in the next Python Polars release).

### PR DESCRIPTION
**Pull request by eitsupi**
Close #371

```python
import polars as pl
from datetime import date
from pyprql import polars_namespace

df = pl.DataFrame({
    "a": [1, 2, 3],
    "b": ["zz", "yy", "xx"],
    "c": [date(1999, 12, 31), date(2010, 10, 10), date(2077, 8, 8)],
})

df.prql.query("select {a, b, c_year = s'EXTRACT(year FROM c)'} | filter a > 1")
# shape: (2, 3)
# ┌─────┬─────┬────────┐
# │ a   ┆ b   ┆ c_year │
# │ --- ┆ --- ┆ ---    │
# │ i64 ┆ str ┆ i32    │
# ╞═════╪═════╪════════╡
# │ 2   ┆ yy  ┆ 2010   │
# │ 3   ┆ xx  ┆ 2077   │
# └─────┴─────┴────────┘
```

### CODE REVIEW

File: pyprql/polars_sub_namespace/prql.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+@pl.api.register_dataframe_namespace("prql")
+@pl.api.register_lazyframe_namespace("prql")
+class PrqlSubNamespace:
+    def __init__(self, df: pl.DataFrame | pl.LazyFrame):
+        self._df = df
+
+    def query(
+        self, prql_query: str, *, table_name: str | None = None
+    ) -> pl.DataFrame | pl.LazyFrame:
```
Comment (eitsupi) [inline pyprql/polars_sub_namespace/prql.py:13]: I think this works fine, but I couldn't figure out how to write the type hint here. Obviously the return value changes depending on the df type, but is there any way to indicate that?
(Perhaps I should inherit the class and define `PrqlDataFrameSubNs` and `PrqlLazyFrameSubNs`?)

**Review by max-sixty — APPROVED** [review]
Looks cool!

No objection to starting this as an optional dependency. We can then promote it if it turns out to be used lots. (I had a bit more objection in #374 to making everything optional).

Another option is to detect whether polars is installed in `pyprql.__init__` and add the namespace without having to import `pyprql.polars_namespace` — depends on the polars norms.

File: pyprql/polars_sub_namespace/prql.py (hunk 1 above)
Comment (max-sixty) [reply pyprql/polars_sub_namespace/prql.py]: Yes if we do something like this:

```
T_Df = TypeVar(pl.DataFrame, pl.LazyFrame)

class PrqlSubNamespace[T_Df]:
    def query(...) -> T_Df
```

...that should work.

Here's an example of a similar impl:
https://github.com/pydata/xarray/blob/d64460795e406bc4a998e2ddae0054a1029d52a9/xarray/core/parallel.py#L229-L235
https://github.com/pydata/xarray/blob/9eb180b7b3df869c21518a17c6fa9eb456551d21/xarray/core/types.py#L154

Comment (eitsupi) [reply pyprql/polars_sub_namespace/prql.py]: Thanks! I will try that.

**Comment by eitsupi** [comment]
> Another option is to detect whether polars is installed in `pyprql.__init__` and add the namespace without having to import `pyprql.polars_namespace` — depends on the polars norms.

That seems like a good idea, but I don't know if it is good practice to do that here.
For now, I would like to merge and release it as is.

**Pull request merged**